druid/examples/hello.rs

```diff
@@ -12,62 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! This is a very small example of how to setup a druid application.
+//! It does the almost bare minimum while still being usefull.
+
+use druid::widget::prelude::*;
 use druid::widget::{Flex, Label, TextBox};
-use druid::{
-    AppLauncher, Data, Env, FontDescriptor, FontFamily, Lens, LocalizedString, UnitPoint, Widget,
-    WidgetExt, WindowDesc,
-};
+use druid::{AppLauncher, WidgetExt, WindowDesc};
 
 const VERTICAL_WIDGET_SPACING: f64 = 20.0;
 const TEXT_BOX_WIDTH: f64 = 200.0;
-const WINDOW_TITLE: LocalizedString<HelloState> = LocalizedString::new("Hello World!");
-
-#[derive(Clone, Data, Lens)]
-struct HelloState {
-    name: String,
-}
 
 pub fn main() {
     // describe the main window
     let main_window = WindowDesc::new(build_root_widget)
-        .title(WINDOW_TITLE)
+        .title("Hello World!")
         .window_size((400.0, 400.0));
 
     // create the initial app state
-    let initial_state = HelloState {
-        name: "World".into(),
-    };
+    let initial_state: String = "World".into();
 
-    // start the application
+    // start the application. Here we pass in the application state.
     AppLauncher::with_window(main_window)
         .launch(initial_state)
         .expect("Failed to launch application");
 }
 
-fn build_root_widget() -> impl Widget<HelloState> {
+fn build_root_widget() -> impl Widget<String> {
     // a label that will determine its text based on the current app data.
-    let label = Label::new(|data: &HelloState, _env: &Env| {
-        if data.name.is_empty() {
+    let label = Label::new(|data: &String, _env: &Env| {
+        if data.is_empty() {
             "Hello anybody!?".to_string()
         } else {
-            format!("Hello {}!", data.name)
+            format!("Hello {}!", data)
         }
     })
-    .with_font(FontDescriptor::new(FontFamily::SERIF).with_size(32.0))
-    .align_horizontal(UnitPoint::CENTER);
+    .with_text_size(32.0);
 
     // a textbox that modifies `name`.
     let textbox = TextBox::new()
         .with_placeholder("Who are we greeting?")
         .with_text_size(18.0)
-        .fix_width(TEXT_BOX_WIDTH)
-        .align_horizontal(UnitPoint::CENTER)
-        .lens(HelloState::name);
```

Comment:
my initial thinking was to use a struct + a lens instead of just `String` in order to highlight the `Lens` trait, which is one of the central concepts in druid? I don't mind changing it though, simple is good..

Reply:
it seems a bit complex for a first look. I think its more important to be able to see how it works first before trying to get people to understand lenses.

Reply:
I think that `Data` and `Lens` are such fundamental concepts in druid that it is worth introducing them immediately.

Reply:
I reverted that back so now its the HelloState again with a lens